src/neuroconv/nwbconverter.py

```diff
@@ -60,15 +60,27 @@ def validate_source(cls, source_data: Dict[str, dict], verbose: bool = True):
         """Validate source_data against Converter source_schema."""
         cls._validate_source_data(source_data=source_data, verbose=verbose)
 
-    def __init__(self, source_data: Dict[str, dict], verbose: bool = True):
+    def __init__(self, source_data: Dict[str, dict] = None, data_interfaces: List = None, verbose: bool = True):
         """Validate source_data against source_schema and initialize all data interfaces."""
         self.verbose = verbose
-        self._validate_source_data(source_data=source_data, verbose=self.verbose)
-        self.data_interface_objects = {
-            name: data_interface(**source_data[name])
-            for name, data_interface in self.data_interface_classes.items()
-            if name in source_data
-        }
+
+        if source_data is not None:
+            self._validate_source_data(source_data=source_data, verbose=self.verbose)
+
+            self.data_interface_objects = {
+                name: data_interface(**source_data[name])
+                for name, data_interface in self.data_interface_classes.items()
+                if name in source_data
+            }
+
+        elif data_interfaces is not None:
+
+            self.data_interface_objects = {
+                f"{interface.__class__.__name__}{index}": interface for index, interface in enumerate(data_interfaces)
+            }
+
+        else:
+            raise IOError("Need to pass either source_data or a list with initialized data interfaces")
```

Comment:
This would be a `ValueError`

Reply:
https://docs.python.org/3/library/exceptions.html#ValueError

Reply:
Thanks, this is no the first time that I get this wrong!